### DISCUSSION TITLE
The Goldendict version 6.4.3-GoldenDict.exe_windows-2019_20230402 doesn't work on Window 11! PLS help

### DISCUSSION THREAD

**Discussion by KamlaK10** [xiaoyifang/goldendict-ng#450] [General]
PLEASE HELP

I am using the GoldenDict , version 1.5.0-RC2-372-gc3ff15f Based on Qt 5.12.3 (GCC 7.3.0, 64 and it is working very smoothly. 
I came across an updated version of Goldendict on github and have been trying to install it but it crashes whenever i search some words, not all the dictionaries are indexed properly. **I have zero knowledge of coding I use use it to learn German**
 
The vesion that I use I downloaded is [6.4.3-GoldenDict.exe_windows-2019_20230402-xapian.zip](https://github.com/xiaoyifang/goldendict/releases/download/v23.04.02-QingmingFestival.230402.ce9436f9/6.4.3-GoldenDict.exe_windows-2019_20230402-xapian.zip)
I even tried downloading this version below but it crashes whenever I some words after a while. And both these version doesn't doesn't many times :/
[6.4.3-GoldenDict.exe_windows-2019_20230402.zip](https://github.com/xiaoyifang/goldendict/releases/download/v23.04.02-QingmingFestival.230402.ce9436f9/6.4.3-GoldenDict.exe_windows-2019_20230402.zip)

**ALSO, pls tell me is this new version really better or just the same like before?** I am sorry if I seem stupid but I hardly use github so I don't understand what I did wrong.

**Comment by xiaoyifang** [comment]
can you upload this information
![image](https://user-images.githubusercontent.com/105986/230251348-6eb8cd0c-9255-4995-b292-4dab9524f35c.png)

And is your platform x64 or arm64?
![image](https://user-images.githubusercontent.com/105986/230252983-c3aeec26-e680-49e8-a5cc-f5fcd07ed254.png)


two options(skip this if too difficult):

1 you can start the goldendict in commandline with ` goldendict.exe --log-to-file`  and upload the log.txt file in the configuration folder which you can open through the help menu.
2 enable the crash dump and upload the crash dmp file for analysis.

**Comment by KamlaK10** [comment]
Sorry for replying late sir/madam
My system's processor and fan was going crazy when I used this new version so I have switched back to the old one. I thank you for your time nevertheless

**Reply by xiaoyifang** [reply]
> My system's processor and fan was going crazy

This depends ,as you have not provide too much information.  You can turn off the fulltext to check again.

If the processor and fan has not stopped for a long time ,this must be a bug ,  if it stops , this should not be a problem.
Pay attention to the status bar of goldendict and verify if there are some fulltext creation processing.

**Reply by KamlaK10** [reply]
Dear Sir/Madam @xiaoyifang 

I am deeply grateful for help and responses, and I would like to apologize for my technological backwardness. I am afraid I don't understand most of the jargons you used like fulltext indexing, crash dump etc. etc.  I use Goldendict to learn words since I am studying German. 
However I have tried my best this time to provide you with the relevant details. Maybe you could figure out the issue :) I am nevertheless indebted for your voluntary help and time. 

**Version information of the Goldendict, which i downloaded**
Pls note- I just use this version because i searched that Xapian version is for developers,. 

Goldendict 23.04.02-QingmingFestival.230402.ce9436f9
windows winnt 10.0.22621 Qt 6.4.3 x86_64-little_endian-llp64
Visual C++ Compiler: 192930148
Flags: MAKE_ZIM_SUPPORT MAKE_EXTRA_TIFF_HANDLER MAKE_CHINESE_CONVERSION_SUPPORT

**System info** 
![image](https://user-images.githubusercontent.com/89730239/231865734-185c9cff-36a8-47d5-a4d8-229b33d9beb8.png)


**crash dump file-**
Note: if this is not the correct file, pls guide me on where I could find this dump file?

https://drive.google.com/drive/folders/12fHAzrNh8ykc6OLu3QNAOa-9Hlw3RKbh?usp=sharing


> 
> If the processor and fan has not stopped for a long time ,this must be a bug , if it stops , this should not be a problem. Pay attention to the status bar of goldendict and verify if there are some fulltext creation processing.

As far as this is concerned, I guess you are asking for this info. This is when the fan and processor were running but it stops after a while.
![indexing](https://user-images.githubusercontent.com/89730239/231868192-aa6e851e-b7fb-489f-b478-708050f30c80.png)

**LASTLY,** this version is prevented from running by my protection software( Mcafee) where the version 1.5.0-RC2-372-gc3ff15f Based on Qt 5.12.3 (GCC 7.3.0, 64 ) is never stopped

Hope all this help!

**Reply by xiaoyifang** [reply]
![image](https://user-images.githubusercontent.com/105986/231916152-53cd3a79-eb5b-422f-897d-8960cd6b328e.png)

this is the full-text indexing ,it is the reason why cpu fans roars.   The cpu roars means it use most of its power to finish this task .
(like a engine of supercar which roars while bicycle can not   :-))

If you do not need this full text search feature,you can disable it in the preference.
![image](https://user-images.githubusercontent.com/105986/231916722-05b60dff-8394-41fc-b8dc-9e5f40fcbaa6.png)

**Reply by KamlaK10** [reply]
oh okay sir/madam
**what about the crash dump data I shared** .  If this is not correct could you tell me please where is this located in windows 11. I just search goldendict in my c drive and it showed me this dump data files

The application always crashes, closes unexpected when I search certain words like Bedeutung, ausführen in german. Whereas i don't face any problem in this version 
![image](https://user-images.githubusercontent.com/89730239/231943202-dbcaaf84-a240-44de-8e8c-7c806001c525.png)

**Reply by xiaoyifang** [reply]
> The application always crashes, closes unexpected when I search certain words like Bedeutung, ausführen in german.

Does this crash can be reproduce ,or it crash everytime when search the above words.
can you share with me the dictionary for investigation. in you google cloud disk.

**Reply by KamlaK10** [reply]
> Does this crash can be reproduce

I don't understand this sir/madam sorry. Could you please reframe this ? 

It crashes every time I search the above words (words like erfolgen, Bedeutung, ausführen etc. ) and even when I try to erase the searched words from this tab 

![image](https://user-images.githubusercontent.com/89730239/231955192-b5b7e35f-c080-461a-bcc0-27ab008b6f27.png)

I already shared the crash reports with you in my drive. I especially searched these words so that it could be refleced in the crash reports uploaded. Maybe you could see that. 
Here is the dictionary version I am using.
https://drive.google.com/drive/folders/1XmlSHpQvmLn3Tmmo-Jgsib69To9mKiOX?usp=sharing
( you don't need the permission this time )
 I have also installed the version too 1.5.0-RC2-372-gc3ff15f Based on Qt 5.12.3 (GCC 7.3.0, 64 ) and using this as a substitute since the Qt 6 x version doesn't work.

**Reply by xiaoyifang** [reply]
> I don't understand this sir/madam sorry. Could you please reframe this ?

Can you share with me the dictionary file ?   the dictionary you are currently using.  not the Goldendict application.

**Reply by KamlaK10** [reply]
the dictionaries I am using are of 2gb worth of storage. I would upload them, but it would take time. I use multiple dictionaries for eng-german-eng

btw it's okay, if all this is too time-consuming and perhaps my system or dictionaries aren't supported by this app, it's my bad luck :)

**Reply by xiaoyifang** [reply]
> the dictionaries I am using are of 2gb worth of storage.

Maybe I have not made this clear .  You do not have to share all your dictionaries with me .
You only need to share the problemic dictionary with me.

This is the steps to find out the problemic dictionary.

![image](https://user-images.githubusercontent.com/105986/232068716-80b963bc-0b1b-4068-a384-5e7247d4928d.png)

**Reply by KamlaK10** [reply]
**There are a few more issues I am encountering!**

 **I have to turnoff my McAfee protection everytime I use Qt6x.**  I had to reinstall this app again because of something , now it doesn't start either.It is taking too long to load.This is not the case with the version that I normally use.  ( The version in the pic is same as quoted above)
![golden dict regular](https://user-images.githubusercontent.com/89730239/232114841-7956936f-0134-4841-93ce-e38065d39437.png)

When the application crashed  after I searched those words( ausführen, bedeutung, erfolgen etc) it wouldn't open. Clicking the app just makes the cursor buffers and it stops then. I tried  running it with it open as  administrator rights too a few times but it didnot response. Only after multiple attempts does the application relaunch.
**here are the most recent crash reports ....**
https://drive.google.com/drive/folders/13nJch7w-18uCuv9UI04CsHDQnts6W3eO?usp=sharing

**And this is the dict which is creating the problem.** 
https://drive.google.com/drive/folders/1efeN1HkfK8EbURghfPUdPrAY3YIpCErU?usp=sharing

However this  dictionary runs fine in the regular version that I use. 

**Another thing that i noticed is there is no extra options for customized scan popup like in previous version.**
![missing options](https://user-images.githubusercontent.com/89730239/232118142-fe6f5eef-45cb-473b-ad3f-658b974c11f5.png)
![no scan options](https://user-images.githubusercontent.com/89730239/232118149-98344f2f-43f8-4725-99da-6be5443a4b76.png)

**Finally please tell me whether the most recent version of Goldendict QT 6x is a heavy-duty software? Will it take a toll on my CPU memory?
 I take classes with the  Goldendict App running in BG, zoom and my pdf application viewer alongside. The Goldendict version that I am using till now doesn't consume soo much memory.**

**Reply by xiaoyifang** [reply]
> **I have to turnoff my McAfee protection everytime I use Qt6x.**

Do not do this ,keep your antivirus software running .  If MacAfee  has affected the running of software (in this case ,the goldendict),
This is a much MacAfee's issue,report this to them.

IMHO,  windows 11 self bundle antivirus should be good enough. 



> When the application crashed  after I searched those words( ausführen, bedeutung, erfolgen etc) it wouldn't open. Clicking the app just makes the cursor buffers and it stops then. I tried  running it with it open as  administrator rights too a few times but it didnot response. Only after multiple attempts does the application relaunch.

The crash issue seems can not reproduce everytime,I  will see what I can do



> Another thing that i noticed is there is no extra options for customized scan popup like in previous version.

The options in Windows has been simplified.
some changes include but not limited to this https://github.com/xiaoyifang/goldendict/pull/207




> Finally please tell me whether the most recent version of Goldendict QT 6x is a heavy-duty software? Will it take a toll on my CPU memory?

Goldendict has embeded a chrome browser engine , it will take a little more memory than original goldendict version.
But the memory should not be too much to affect your other applications and system.

It is not a heavy-duty software,  Apart from full text creation it has no other background tasks(Which I think you have disabled).  After search it will release the cpu as well.

**Reply by xiaoyifang** [reply]
> When the application crashed  after I searched those words( ausführen, bedeutung, erfolgen etc) it wouldn't open. Clicking the app just makes the cursor buffers and it stops then. I tried  running it with it open as  administrator rights too a few times but it didnot response. Only after multiple attempts does the application relaunch.

I guess there should be some unexit process in the task manager,  you can open your task manager and kill goldendict.exe 's process if any.

**Comment by xiaoyifang** [comment]
https://user-images.githubusercontent.com/105986/232174293-08386949-0516-438b-8661-13b2ee6a5cfe.mp4

I can not reproduce this anymore.

Can you create a seperate dictionary group to hold this dictionary and verify again?
The issue may come from another dictionary.

**Reply by KamlaK10** [reply]
Surprisingly enough now everything is  fine except the sluggish loading. That dictionary is working fine. I swear, though, I had reinstalled the programme twice and each time it crashed initially until I did what you asked me like separately checking each dictionary.
Pls see the video to get an idea about the loading process

https://user-images.githubusercontent.com/89730239/232230917-057ddddb-d4c4-47a3-a276-39d4494ea126.mp4

**Reply by KamlaK10** [reply]
**Could you pls also guide me about the anki integration process?** It is stated on the description page that in this version of Goldendict it is possible to integrate with Anki. Like is there **any help topic dedicated to that** I will read and follow the steps myself.  That's the main reason actually why I switched to newer version, so your help would be greatly appreciated.

**Reply by xiaoyifang** [reply]
https://github.com/xiaoyifang/goldendict/blob/staged/howto/how%20to%20connect%20with%20anki.md

**Reply by xiaoyifang** [reply]
btw, from your video ,I can see that you still have your fulltext enabled.

**Reply by xiaoyifang** [reply]
> Surprisingly enough now everything is fine except the sluggish loading. T

I have also no idea,  maybe try to add goldendict as an exclusion to your antivirus and check again.

**Reply by KamlaK10** [reply]
> btw, from your video ,I can see that you still have your fulltext enabled.

Yes because I use dictionaries of different formats like some are mdict, slob, dsl etc. etc. So if I turn everything off I wouldn't be able to search them sir/madam.  
And thanks a lot for your support thus far.

**Reply by xiaoyifang** [reply]
You mean you also use fulltext search in the time. Ctrl+Shift+F.
Even disable fulltext ,all the dictionaries basic query function still  can be used .  

And you are very kind to call me sir, you can call me xiaoyifang.

**Reply by xiaoyifang** [reply]
> > Surprisingly enough now everything is fine except the sluggish loading. T
> 
> I have also no idea, maybe try to add goldendict as an exclusion to your antivirus and check again.

If the white sluggish loading keeps coming ,it may indicates an hidden bug.  keep watch.
If it can not start in the first time,  look your taskmanager to check if there are some lingering process `goldendict.exe`

**Comment by xiaoyifang** [comment]
When this issue happens ,do you happen to quit the official old webkit goldendict just a short time ago?

**Reply by xiaoyifang** [reply]
If crashes happen again. use the following release .
https://github.com/xiaoyifang/goldendict-ng/releases/tag/v23.04.03-alpha.230501.a7d6c45f

 try to reproduce the crash ,there will be an dmp file alongside with goldendict.exe in folder `crash`.

All the following qt6-windows-xapian packages have built with google breakpad support ,which enable to catch the dump and analysis the root cause of the crash.
---------------
